File: ui-ngx/src/app/core/http/trendz-settings.service.ts
// SPDX-FileCopyrightText: Copyright The Thingsboard Authors
// SPDX-License-Identifier: Apache-2.0
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { TrendzSettings } from '@shared/models/trendz-settings.models';
import { defaultHttpOptionsFromConfig, RequestConfig } from '@core/http/http-utils';

@Injectable({
  providedIn: 'root'
})
export class TrendzSettingsService {

  constructor(
    private http: HttpClient
  ) {}

  public getTrendzSettings(config?: RequestConfig): Observable<TrendzSettings> {
    return this.http.get<TrendzSettings>(`/api/trendz/settings`, defaultHttpOptionsFromConfig(config))
  }

  public saveTrendzSettings(trendzSettings: TrendzSettings, config?: RequestConfig): Observable<TrendzSettings> {
    return this.http.post<TrendzSettings>(`/api/trendz/settings`, trendzSettings, defaultHttpOptionsFromConfig(config))
  }
}

File: common/dao-api/src/main/java/org/thingsboard/server/dao/trendz/TrendzSettingsService.java
// SPDX-FileCopyrightText: Copyright The Thingsboard Authors
// SPDX-License-Identifier: Apache-2.0
package org.thingsboard.server.dao.trendz;

import org.thingsboard.server.common.data.id.TenantId;
import org.thingsboard.server.common.data.trendz.TrendzSettings;

public interface TrendzSettingsService {

    void saveTrendzSettings(TenantId tenantId, TrendzSettings settings);

    TrendzSettings findTrendzSettings(TenantId tenantId);

    void deleteTrendzSettings(TenantId tenantId);

}
